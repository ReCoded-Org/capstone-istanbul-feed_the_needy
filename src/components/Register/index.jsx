import React, { useState } from "react";
import { Form, Input, Button, Checkbox, message } from "antd";
import { useHistory } from "react-router-dom";
import firebase from "../../firebaseConfig";
import { useTranslation } from "react-i18next";
import { MailOutlined } from "@ant-design/icons";
import "./index.css";
import { Layout } from "../Layout";

const db = firebase.firestore();
const layout = Layout.layout;
const tailLayout = Layout.tailLayout;

const Register = ({ isTesting }) => {
  const { t } = useTranslation();
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [name, setName] = useState("");

  const addOrganization = () => {
    firebase.auth().onAuthStateChanged((user) => {
      if (user) {
        db.collection("restaurants").doc(user.uid).set({
          name: name,
          email: email,
          uid: user.uid,
          website: "",
          description: "",
          logo: "",
          logoURL: "",
          logoName: "",
        });
        setName("");
      }
    });
  };

  const handleClick = () => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(() => {
        message.success(t("register.registerTitle.registerSuccessMessage"));
      })
      .catch((err) => {
        console.log("err", err);
        message.warning(err.message);
      });
  };

  let history = useHistory();

  const routeChange = () => {
    let path = "/admin";
    history.push(path);
  };

  const addDetails = (e) => {
    handleClick();
    addOrganization();
    routeChange();
  };

  return (
    <div className="formInputSection" {...layout} {...tailLayout}>
      <Form onFinish={addDetails}>
        <Form.Item>
          <Input
            onChange={(e) => setName(e.target.value)}
            value={name}
            placeholder={t("register.registerTitle.registerTitle4")}
            action="click"
            type="text"
            className="ant-input"
          />
        </Form.Item>
        <Form.Item
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          rules={[{ required: true, message: "Please input your email!" }]}
        >
          <Input
            prefix={<MailOutlined />}
            placeholder={t("register.registerTitle.registerTitle1")}
            action="click"
            type="email"
            className="ant-input"
          />
        </Form.Item>
        <Form.Item
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          rules={[{ required: true, message: "Please input your password!" }]}
        >
          <Input
            placeholder={t("register.registerTitle.registerTitle2")}
            action="click"
            type="password"
            className="ant-input"
          />
        </Form.Item>
        <Form.Item
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          rules={[{ required: true, message: "Please input your password!" }]}
        >
          <Input
            placeholder={t("register.registerTitle.registerTitle2")}
            action="click"
            type="password"
            className="ant-input"
          />
        </Form.Item>
        <Form.Item valuePropName="checked">
          <Checkbox type="checkbox" required>
            {t("register.registerTitle.registerTitle3")}
          </Checkbox>
        </Form.Item>
        <Form.Item>
          {isTesting ? (
            <button>Donate</button>
          ) : (
            <Button
              style={{ background: "#2F80ED" }}
              type="submit"
              htmlType="submit"
              className="loginButton"
            >
              {t("register.registerTitle.registerTitle0")}
            </Button>
          )}
        </Form.Item>
      </Form>
    </div>
  );
};

export default Register;
