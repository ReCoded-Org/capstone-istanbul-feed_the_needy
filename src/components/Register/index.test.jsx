import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Register from "./index";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const isTesting = true;
it("Login renders correctly", () => {
  const tree = renderer.create(<Register isTesting={isTesting} />).toJSON();
  expect(tree).toMatchSnapshot();
});
