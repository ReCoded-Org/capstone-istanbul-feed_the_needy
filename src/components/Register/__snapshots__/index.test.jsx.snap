// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="formInputSection"
  labelCol={
    Object {
      "span": 4,
    }
  }
  wrapperCol={
    Object {
      "offset": 8,
      "span": 16,
    }
  }
>
  <form
    className="ant-form ant-form-horizontal"
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              action="click"
              className="ant-input ant-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="register.registerTitle.registerTitle4"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      onChange={[Function]}
      style={Object {}}
      value=""
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-affix-wrapper ant-input"
              onMouseUp={[Function]}
            >
              <span
                className="ant-input-prefix"
              >
                <span
                  aria-label="mail"
                  className="anticon anticon-mail"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="mail"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                    />
                  </svg>
                </span>
              </span>
              <input
                action="click"
                className="ant-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="register.registerTitle.registerTitle1"
                style={null}
                type="email"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      onChange={[Function]}
      style={Object {}}
      value=""
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              action="click"
              className="ant-input ant-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="register.registerTitle.registerTitle2"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      onChange={[Function]}
      style={Object {}}
      value=""
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              action="click"
              className="ant-input ant-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="register.registerTitle.registerTitle2"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <label
              className="ant-checkbox-wrapper"
            >
              <span
                className="ant-checkbox"
                style={Object {}}
              >
                <input
                  checked={false}
                  className="ant-checkbox-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="checkbox"
                />
                <span
                  className="ant-checkbox-inner"
                />
              </span>
              <span>
                register.registerTitle.registerTitle3
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <button>
              Donate
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
